### PR TITLE
BLEND-181: Improve installing render delegate

### PR DESCRIPTION
**Pull request by VascoPi**
Improved Addon properties UI.
Added Delegate directory settings.
Made delegate list dynamic.

### CODE REVIEW

File: extern/usdhydra/addon/properties/preferences.py [hunk 1]
```diff
@@ -89,6 +96,12 @@ def update_log_level(self, context):
         name="Render Delegate",
         default=False,
     )
+    settings: EnumProperty(
+        name="",
+        items=(('SETTINGS', "Settings", "aaa"),
```
Comment (bnagirniak) [inline extern/usdhydra/addon/properties/preferences.py]: fix "aaa"

File: extern/usdhydra/addon/utils/delegate.py [hunk 2]
```diff
@@ -42,16 +31,26 @@ def set_progress(self, msg):
         self.progress = msg
         update_ui(area_type='PREFERENCES')
 
+    def update_delegates(self):
+        self.delegates = session_get_render_plugins()
+
+    def register_delegates(self):
+        from ..engine import init, exit
+        exit()
```
Comment (bnagirniak) [inline extern/usdhydra/addon/utils/delegate.py]: better: `from .. import engine`

File: extern/usdhydra/python.cpp [hunk 3]
```diff
@@ -37,14 +37,10 @@ static PyObject *init_func(PyObject * /*self*/, PyObject *args)
...
-  
-  std::string env("PATH=");
-  env += delegates_dir_str + "/lib;";
-  env += getenv("PATH");
-  putenv(env.c_str());
-  
+
   for (pxr::PlugPluginPtr p : registry.GetAllPlugins()) {
     printf("%s %s\n", p->GetName().c_str(), p->GetPath().c_str());
   }
```
Comment (bnagirniak) [inline extern/usdhydra/python.cpp]: remove this

File: extern/usdhydra/addon/properties/preferences.py [hunk 4]
```diff
@@ -99,40 +112,40 @@ def _draw_settings(layout):
...
 
-            _draw_delegate_item(layout)
-            layout.separator()
+            if manager.delegates is None:
+                manager.update_delegates()
 
-        def _draw_delegate_item(layout):
-            for key, value in get_delegates().items():
-                row = layout.row(align=True)
+            plugins = Plug.Registry().GetAllPlugins()
```
Comment (bnagirniak) [inline extern/usdhydra/addon/properties/preferences.py]: remove this

**Review by bnagirniak — CHANGES_REQUESTED** [review]

**Review by DagerD — APPROVED** [review]

**Review by bnagirniak — APPROVED** [review]

**Pull request merged**